### DISCUSSION TITLE
Missing Chocolatey packager for native image

### DISCUSSION THREAD

**Discussion by mthmulders** [jreleaser/jreleaser#601] [Q&A] [answered]
I'm using JReleaser to release a native executable (built with GraalVM). Publication with Snap and Homebrew works fine, but I can't get Chocolatey to work yet. I have a `<appname>-bucket` repository on Github with a GitHub action similar to that of [JReleaser](https://github.com/jreleaser/chocolatey-bucket), that would publish the spec to Chocolatey.

But that bucket isn't populated by JReleaser. I thought it might be because there's no **META-INF/jreleaser/templates/native-image/chocolatey** template in [jreleaser-templates](https://github.com/jreleaser/jreleaser/tree/main/core/jreleaser-templates). So I [added](https://github.com/mthmulders/mcs/commit/d030687d7b1dfbe28b79442b90faab90f873101d) it to my project, hoping it would populate the Chocolatey bucket. That didn't work as expected.

Now I'm a bit lost as to why the bucket isn't populated... Any thoughts?

**Comment by aalmiray** [comment, marked as answer]
Chocolatey is currently not supported by the `NATIVE_IMAGE` distribution type https://jreleaser.org/guide/latest/distributions/native-image.html#_packager_support

This is a missing feature that requires figuring out the correct settings for a Windows distribution, assuming `x86_64` as the only choice at the moment, unless chocolatey has a setting for Windows on ARM.

**Reply by aalmiray** [reply]
This being said, the `JLINK` distribution does support chocolatey so perhaps there's not much to do other than adding the proper template and validation.

**Reply by aalmiray** [reply]
Filed and fixed https://github.com/jreleaser/jreleaser/issues/602

Be aware that you'll have to update your own chocolatey templates as the expected executable should have `.exe` instead of `.bat` extension. Also, JReleaser can't publish the package for you, you'll have to create an account first then submit the package. This is how JReleaser does it for itself https://github.com/jreleaser/chocolatey-bucket/blob/main/.github/workflows/push.yml

**Reply by mthmulders** [reply]
Wow, thanks (again!). That's a really quick answer. Just to make sure, once I'm on JReleaser 0.10.0, I could simply remove the custom Chocolatey templates I have now and take those that come with JReleaser itself. They don't have `.bat` anywhere, as far as I can see...

I've already prepared the "push to Chocolatey registry" action (just like the one that JReleaser uses itself) so I guess I'll be good to go... :-)

**Reply by aalmiray** [reply]
I thought in .ps1 files inside tools make a reference to a .bat files, don't they?

**Reply by mthmulders** [reply]
Your commit [a5512623](https://github.com/jreleaser/jreleaser/commit/a5512623610c0578a58e08c58cabdc36c4f092e9) does not refer bat files AFAICS. My custom templates do, but I plan to remove them as soon as JReleaser 0.10.0 sees the light.
